web/api/v1/api_test.go

```diff
@@ -1354,7 +1354,7 @@ func testEndpoints(t *testing.T, api *API, tr *testTargetRetriever, testLabelAPI
 							alertingRule{
 								State:       "inactive",
 								Name:        "test_metric4",
-								Query:       "up == 1",
+								Query:       "up== 1",
```

Comment:
But why would the test case change here?

Reply:
`==` is considered as `EQL` item by the lexer. This is good. But, `=` is also considered as `EQL`. When used in labels, it should result in no spaces in labels like `<label_name>=<label_value>`. Now here comes the issue. This will mean that where ever you encounter EQL, don't leave spaces before and after an EQL. 

Now in case of `up == 1`, up is IDENTIFIER. `==` is `EQL`, so it is printed as if it is the case of a label_matchers. And, `1` is numeric literal, so it has space before itself.

I hope, this makes sense.

Reply:
> don't leave spaces before and after an EQL.

Right, the `after` is missing here. Can we tie the spacing logic to the `EQL`/`ASSIGN` items and not the RHS so that it is same on both sides always? `up== 1` is not pretty :)

Reply:
Ahh, I don't think how reliable that will be. But yes, whenever we encounter EQL, we can check if the value is `=` or `==` and based on the value, it will work perfectly. But then, the person (I can take it if we recognize that as a bug) who solves that issue should make changes here when that issue is fixed.

Reply:
We can fix that spacing with the bug fix later. What I meant to say is we avoid `And, 1 is numeric literal, so it has space before itself`, which sounds like we look at the number and add a space. Rather, see that it's EQL/ASSIGN and either add spaces on both sides or dont add at all.

Reply:
These are fixed now. Hence, resolving.